Reed/Components/DefinableText/DefinableText.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  DefinableText.swift
+//  Reed
+//
+//  Created by Hugo Zhan on 11/3/20.
+//  Copyright © 2020 Roger Luo. All rights reserved.
+//
+
+import SwiftUI
+import UIKit
+
+struct Term: Equatable, Hashable, Identifiable {
+    var id = UUID()
+    var reading: String
+    var term: String
+}
+
+struct Definition: Equatable, Hashable, Identifiable {
+    var id = UUID()
+    var specicificLexemes: String
+    var definition: String
+}
+
+struct DefinitionDetails: Equatable, Hashable, Identifiable {
+    var id = UUID()
+    var title: String
+    var primaryReading: String
+    var terms: [Term]
+    var definitions: [Definition]
+    
+    static func == (lhs: DefinitionDetails, rhs: DefinitionDetails) -> Bool {
+        return lhs.definitions == rhs.definitions
+    }
+}
+
+struct DefinableText: UIViewRepresentable {
+    @Binding var content: NSMutableAttributedString
+    @ObservedObject var viewModel: DefinableTextViewModel
```

Comment:
Do we still need `@ObservedObject`? If so, should it be `@StateObject`?

Reply:
removed